lib/interfaces/index.ts

```diff
@@ -0,0 +1,10 @@
+export interface IImpactModelInterface {
+    modelIdentifier(): string
+
+    // params is a reserved keyword in C#. Hence it can not be used.
+    configure(name: string, staticParams: object | undefined): Promise<IImpactModelInterface>
```

Comment:
You mentioned on our last slack conversation that function signatures can be of the form "[key: string]: any", but I see them as objects again here.

Reply:
I am using it in internal calls. Since most of the code which was in the calculate methods earlier is refactored to separate methods, it sufficient for those methods to have the index signature.